Introducing `SimOnGen` and `GenOnly` wfs for 2024 and 2026D110

AdrianoDee opened the pull request:
This PR proposes the addition of:
- `Sim` only step running only the SIM step (to be run on top of GEN);
- `SimOnGen`  wfs running the SIM step separated from `GEN` for 2024 and 2026D110 conditions;
- `GenOnly` wfs running only the GEN step for 2024 and 2026D110 conditions.

As a bonus the `ALCA` input for "upgrade" workflows is updated in order to be more general and to take the output of the `RECO` step rather than being hardcoded to be `step3.root`.

These are useful for us (PdmV) e.g.:

1. to run RelVal jobs recycling only the GEN rerunning the SIM. This will help when we have changes in simulation (e.g. changes in Phase2 geometries) than needs to be validated allowing us to reduce the statistical fluctuations recycling the underlying GEN.
2. to run GEN only wfs (for MinBias, or default samples) without having to act manually on each workflow.

#### PR Validation

Running, e.g., `15824.0` and `15634.0`.

cmsbuild commented:
cms-bot internal usage<!-- bot cache: {"commits":{"a327e4aa775ff8279d7c475536b138330c3e5d1b":{"files":["Configuration/PyReleaseValidation/python/relval_steps.py","Configuration/PyReleaseValidation/python/relval_upgrade.py","Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py"],"squashed":false,"time":1715955992}},"emoji":{},"last_seen_sha":"a327e4aa775ff8279d7c475536b138330c3e5d1b","signatures":{}} -->

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-45005/40287

- This PR adds an extra 76KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/PyReleaseValidation/python/relval_steps.py  modified in PR(s): #44964, #44965
  - File  Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py  modified in PR(s): #33532, #43761, #43964, #44576, #44785, #44964

cmsbuild commented:
A new Pull Request was created by  @AdrianoDee for master.

It involves the following packages:

- Configuration/PyReleaseValidation (**pdmv, upgrade**)


@srimanob, @cmsbuild, @sunilUIET, @miquork, @AdrianoDee, @subirsarkar can you please review it and eventually sign? Thanks.
@Martin-Grunewald, @makortel, @missirol, @fabiocos, @slomeo this is something you requested to watch as well.
@rappoccio, @sextonkennedy, @antoniovilela you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

AdrianoDee commented:
test parameters:
- workflows=15824.0,15634.0
- workflow_opts = -w upgrade

AdrianoDee commented:
please test

cmsbuild commented:
-1

**Failed Tests**: RelVals 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39443/summary.html
**COMMIT**: a327e4aa775ff8279d7c475536b138330c3e5d1b
**CMSSW**: CMSSW_14_1_X_2024-05-20-2300/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/45005/39443/install.sh` to create a dev area with all the needed externals and cmssw changes.


* **DAS Queries**: The DAS query tests failed, see the summary page for details.


## RelVals

- [15824.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39443/runTheMatrix-results/15824.0_TTbar_13+2024SimOnGen)
```
----- Begin Fatal Exception 21-May-2024 14:26:00 CEST-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing  Event run: 1 lumi: 1 event: 1 stream: 0
   [1] Running path 'pathALCARECOEcalESAlign'
   [2] Calling method for module TrackSelector/'esSelectedTracks'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for type: std::vector<reco::Track>
Looking for module label: generalTracks
Looking for productInstanceName: 

   Additional Info:
      [a] If you wish to continue processing events after a ProductNotFound exception,
add "TryToContinue = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.

----- End Fatal Exception -------------------------------------------------
```

Commit:
New GenOnly and SimOnGen wfs:
- Adding Sim step.
- Introducing GenOnly and SimOnGen wfs for 2024 and D110.
- Generalizing ALCA inputs.

AdrianoDee commented:
Note: I've added a note in the description about the needed update to the `ALCA` input coming from `RECO` step (hardcoded in `master`, generalized in this PR).

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-45005/40291

- This PR adds an extra 64KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/PyReleaseValidation/python/relval_steps.py  modified in PR(s): #44964, #44965
  - File  Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py  modified in PR(s): #33532, #43761, #43964, #44576, #44785, #44964

cmsbuild commented:
Pull request #45005 was updated. @cmsbuild, @subirsarkar, @sunilUIET, @miquork, @AdrianoDee, @srimanob can you please check and sign again.

cmsbuild commented:
-1

**Failed Tests**: RelVals 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39454/summary.html
**COMMIT**: 75bcbca90ed53d8a35c30b1288b0ffb4e17a00cb
**CMSSW**: CMSSW_14_1_X_2024-05-20-2300/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/45005/39454/install.sh` to create a dev area with all the needed externals and cmssw changes.


* **DAS Queries**: The DAS query tests failed, see the summary page for details.


## RelVals

- [4.53](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39454/runTheMatrix-results/4.53_RunPhoton2012B)```4.53_RunPhoton2012B/step3_RunPhoton2012B.log```
- [139.001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39454/runTheMatrix-results/139.001_RunMinimumBias2021)```139.001_RunMinimumBias2021/step3_RunMinimumBias2021.log```
- [136.731](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39454/runTheMatrix-results/136.731_RunSinglePh2016B)```136.731_RunSinglePh2016B/step3_RunSinglePh2016B.log```
<details>
<summary>Expand to see more relval errors ...</summary>

- [136.874](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39454/runTheMatrix-results/136.874_RunEGamma2018C)
- [136.793](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39454/runTheMatrix-results/136.793_RunDoubleEG2017C)
- [4.22](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39454/runTheMatrix-results/4.22_RunCosmics2011A)
- [7.3](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39454/runTheMatrix-results/7.3_CosmicsSPLoose2018)
- [8.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39454/runTheMatrix-results/8.0_BeamHalo)
- [25.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39454/runTheMatrix-results/25.0_TTbar)
- [1001.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39454/runTheMatrix-results/1001.0_RunMinBias2011A)
- [11634.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39454/runTheMatrix-results/11634.0_TTbar_14TeV+2021)
- [12434.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39454/runTheMatrix-results/12434.0_TTbar_14TeV+2023)
- [12834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39454/runTheMatrix-results/12834.0_TTbar_14TeV+2024)
- [12846.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39454/runTheMatrix-results/12846.0_ZEE_14+2024)
- [24900.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39454/runTheMatrix-results/24900.0_CloseByPGun_CE_H_Coarse_Scint+2026D98)
- [24896.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39454/runTheMatrix-results/24896.0_CloseByPGun_CE_E_Front_120um+2026D98)
- [24834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39454/runTheMatrix-results/24834.0_TTbar_14TeV+2026D98)
- [23234.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39454/runTheMatrix-results/23234.0_TTbar_14TeV+2026D94)
- [24834.911](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39454/runTheMatrix-results/24834.911_TTbar_14TeV+2026D98_DD4hep)
- [15824.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39454/runTheMatrix-results/15824.0_TTbar_13+2024SimOnGen)
</details>

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-45005/40301

- This PR adds an extra 80KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/PyReleaseValidation/python/relval_steps.py  modified in PR(s): #44964, #44965
  - File  Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py  modified in PR(s): #33532, #43761, #43964, #44785, #44964

cmsbuild commented:
Pull request #45005 was updated. @srimanob, @AdrianoDee, @miquork, @cmsbuild, @subirsarkar, @sunilUIET can you please check and sign again.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-45005/40340

- This PR adds an extra 76KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/PyReleaseValidation/python/relval_steps.py  modified in PR(s): #44965
  - File  Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py  modified in PR(s): #33532, #43761, #43964, #44785, #44910

cmsbuild commented:
Pull request #45005 was updated. @subirsarkar, @sunilUIET, @srimanob, @cmsbuild, @AdrianoDee, @miquork can you please check and sign again.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-45005/40373

- This PR adds an extra 80KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/PyReleaseValidation/python/relval_steps.py  modified in PR(s): #44965, #45055
  - File  Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py  modified in PR(s): #33532, #43761, #43964, #44785, #44910

cmsbuild commented:
Pull request #45005 was updated. @srimanob, @miquork, @AdrianoDee, @sunilUIET, @subirsarkar, @cmsbuild can you please check and sign again.

AdrianoDee commented:
test parameters:
- workflows=31834.0,16034.0,16234.0,31634.0
- workflow_opts = -w upgrade

AdrianoDee commented:
please test

cmsbuild commented:
-1

**Failed Tests**: RelVals 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39552/summary.html
**COMMIT**: d39f5b688a1ea564b5ce6a93c48f686a432860f6
**CMSSW**: CMSSW_14_1_X_2024-05-27-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/45005/39552/install.sh` to create a dev area with all the needed externals and cmssw changes.


## RelVals

- [4.53](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39552/runTheMatrix-results/4.53_RunPhoton2012B)```4.53_RunPhoton2012B/step3_RunPhoton2012B.log```
- [139.001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39552/runTheMatrix-results/139.001_RunMinimumBias2021)```139.001_RunMinimumBias2021/step3_RunMinimumBias2021.log```
- [136.874](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39552/runTheMatrix-results/136.874_RunEGamma2018C)```136.874_RunEGamma2018C/step3_RunEGamma2018C.log```
<details>
<summary>Expand to see more relval errors ...</summary>

- [136.731](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39552/runTheMatrix-results/136.731_RunSinglePh2016B)
- [136.793](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39552/runTheMatrix-results/136.793_RunDoubleEG2017C)
- [4.22](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39552/runTheMatrix-results/4.22_RunCosmics2011A)
- [7.3](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39552/runTheMatrix-results/7.3_CosmicsSPLoose2018)
- [8.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39552/runTheMatrix-results/8.0_BeamHalo)
- [25.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39552/runTheMatrix-results/25.0_TTbar)
- [1001.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39552/runTheMatrix-results/1001.0_RunMinBias2011A)
- [11634.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39552/runTheMatrix-results/11634.0_TTbar_14TeV+2021)
- [12434.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39552/runTheMatrix-results/12434.0_TTbar_14TeV+2023)
- [12834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39552/runTheMatrix-results/12834.0_TTbar_14TeV+2024)
- [12846.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39552/runTheMatrix-results/12846.0_ZEE_14+2024)
- [24896.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39552/runTheMatrix-results/24896.0_CloseByPGun_CE_E_Front_120um+2026D98)
- [24900.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39552/runTheMatrix-results/24900.0_CloseByPGun_CE_H_Coarse_Scint+2026D98)
- [24834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39552/runTheMatrix-results/24834.0_TTbar_14TeV+2026D98)
- [23234.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39552/runTheMatrix-results/23234.0_TTbar_14TeV+2026D94)
- [24834.911](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39552/runTheMatrix-results/24834.911_TTbar_14TeV+2026D98_DD4hep)
- [16234.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39552/runTheMatrix-results/16234.0_TTbar_14TeV+2024SimOnGen)
- [31834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39552/runTheMatrix-results/31834.0_TTbar_14TeV+2026D110SimOnGen)
</details>

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-45005/40417

- This PR adds an extra 80KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/PyReleaseValidation/python/relval_steps.py  modified in PR(s): #44965, #45055
  - File  Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py  modified in PR(s): #33532, #43761, #43964, #44785, #44910

cmsbuild commented:
Pull request #45005 was updated. @miquork, @subirsarkar, @srimanob, @cmsbuild, @sunilUIET, @AdrianoDee can you please check and sign again.

AdrianoDee commented:
please test

cmsbuild commented:
-1

**Failed Tests**: RelVals 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39606/summary.html
**COMMIT**: 06b719108e0c2619cfa2d1e4f57b7b5c98944fa0
**CMSSW**: CMSSW_14_1_X_2024-05-29-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/45005/39606/install.sh` to create a dev area with all the needed externals and cmssw changes.


## RelVals

- [4.53](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39606/runTheMatrix-results/4.53_RunPhoton2012B)```4.53_RunPhoton2012B/step3_RunPhoton2012B.log```
- [139.001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39606/runTheMatrix-results/139.001_RunMinimumBias2021)```139.001_RunMinimumBias2021/step3_RunMinimumBias2021.log```
- [136.731](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39606/runTheMatrix-results/136.731_RunSinglePh2016B)```136.731_RunSinglePh2016B/step3_RunSinglePh2016B.log```
<details>
<summary>Expand to see more relval errors ...</summary>

- [136.793](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39606/runTheMatrix-results/136.793_RunDoubleEG2017C)
- [136.874](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39606/runTheMatrix-results/136.874_RunEGamma2018C)
- [7.3](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39606/runTheMatrix-results/7.3_CosmicsSPLoose2018)
- [25.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39606/runTheMatrix-results/25.0_TTbar)
</details>

Commit:
Restricting GenOnly wfs to baseline

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-45005/40450

- This PR adds an extra 80KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/PyReleaseValidation/python/relval_steps.py  modified in PR(s): #44965, #45016, #45055
  - File  Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py  modified in PR(s): #33532, #43761, #43964, #44785, #44910

cmsbuild commented:
Pull request #45005 was updated. @subirsarkar, @srimanob, @sunilUIET, @cmsbuild, @AdrianoDee, @miquork can you please check and sign again.

AdrianoDee commented:
test parameters:
- workflows= 16034.0, 16000.0, 31634.0, 16234.0
- workflow_opts = -w upgrade

AdrianoDee commented:
please test

cmsbuild commented:
-1

**Failed Tests**: RelVals 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39699/summary.html
**COMMIT**: 86a8712c03d091e947c8eaa37e0d26708f73ecea
**CMSSW**: CMSSW_14_1_X_2024-06-04-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/45005/39699/install.sh` to create a dev area with all the needed externals and cmssw changes.


## RelVals

- [11634.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39699/runTheMatrix-results/11634.0_TTbar_14TeV+2021)
```
----- Begin Fatal Exception 04-Jun-2024 18:54:43 CEST-----------------------
An exception of category 'FileOpenError' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing input source of type PoolSource
   [2] Calling RootInputFileSequence::initTheFile()
   [3] Calling StorageFactory::open()
   [4] Calling File::sysopen()
Exception Message:
Failed to open the file 'step4.root'
   Additional Info:
      [a] Input file file:step4.root could not be opened.
      [b] open() failed with system error 'No such file or directory' (error code 2)
----- End Fatal Exception -------------------------------------------------
```
- [12434.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39699/runTheMatrix-results/12434.0_TTbar_14TeV+2023)
```
----- Begin Fatal Exception 04-Jun-2024 18:55:46 CEST-----------------------
An exception of category 'FileOpenError' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing input source of type PoolSource
   [2] Calling RootInputFileSequence::initTheFile()
   [3] Calling StorageFactory::open()
   [4] Calling File::sysopen()
Exception Message:
Failed to open the file 'step4.root'
   Additional Info:
      [a] Input file file:step4.root could not be opened.
      [b] open() failed with system error 'No such file or directory' (error code 2)
----- End Fatal Exception -------------------------------------------------
```
- [12834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39699/runTheMatrix-results/12834.0_TTbar_14TeV+2024)
```
----- Begin Fatal Exception 04-Jun-2024 18:58:55 CEST-----------------------
An exception of category 'FileOpenError' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing input source of type PoolSource
   [2] Calling RootInputFileSequence::initTheFile()
   [3] Calling StorageFactory::open()
   [4] Calling File::sysopen()
Exception Message:
Failed to open the file 'step4.root'
   Additional Info:
      [a] Input file file:step4.root could not be opened.
      [b] open() failed with system error 'No such file or directory' (error code 2)
----- End Fatal Exception -------------------------------------------------
```
<details>
<summary>Expand to see more relval errors ...</summary>

- [12846.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39699/runTheMatrix-results/12846.0_ZEE_14+2024)
- [24900.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39699/runTheMatrix-results/24900.0_CloseByPGun_CE_H_Coarse_Scint+2026D98)
- [24896.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39699/runTheMatrix-results/24896.0_CloseByPGun_CE_E_Front_120um+2026D98)
- [24834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39699/runTheMatrix-results/24834.0_TTbar_14TeV+2026D98)
- [23234.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39699/runTheMatrix-results/23234.0_TTbar_14TeV+2026D94)
- [24834.911](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39699/runTheMatrix-results/24834.911_TTbar_14TeV+2026D98_DD4hep)
- [16234.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbc0ba/39699/runTheMatrix-results/16234.0_TTbar_14TeV+2024SimOnGen)
</details>

Commit:
Fix ALCA input

AdrianoDee commented:
please test

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-45005/40497

- This PR adds an extra 20KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/PyReleaseValidation/python/relval_steps.py  modified in PR(s): #45055
  - File  Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py  modified in PR(s): #33532, #43761, #43964, #44785, #44910, #45117

cmsbuild commented:
Pull request #45005 was updated. @sunilUIET, @srimanob, @subirsarkar, @AdrianoDee, @miquork can you please check and sign again.

AdrianoDee commented:
+pdmv
(new wfs run as expected)

srimanob commented:
+Upgrade

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @sextonkennedy, @antoniovilela, @rappoccio (and backports should be raised in the release meeting by the corresponding L2)

rappoccio commented:
+1

smuzaffar commented:
@AdrianoDee , workflow 105.0 is failing in IB [a].  May be change in this PR has broken it? Can you lease check

[a] https://cmssdt.cern.ch/SDT/html/cmssdt-ib/#/relVal/CMSSW_14_1/2024-06-12-2300?selectedArchs=el8_amd64_gcc12&selectedFlavors=X&selectedStatus=failed
```
usage: cmsDriver.py <TYPE> [options].
Example:

cmsDriver.py reco -s RAW2DIGI,RECO --conditions STARTUP_V4::All --eventcontent RECOSIM
cmsDriver.py: error: argument --filein: expected one argument
```

smuzaffar commented:
I have proposed a fix https://github.com/cms-sw/cmssw/pull/45214

srimanob commented on Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:
```diff
@@ -3133,6 +3152,21 @@ def condition(self, fragment, stepList, key, hasHarvest):
     },
+    '2026D110GenOnly' : {
+        'Geom' : 'Extended2026D110',
+        'Beamspot' : 'HLLHC',
```
Hi @AdrianoDee 
Just wonder that we will have the right output with this setting of beamspot?

AdrianoDee replied:
Hi Phat, I see this message only now. Isn't it just dummy it here? I mean: the BS is defined at SIM step no? Or we have something already at GEN step?

srimanob replied:
Hi @AdrianoDee 
Thanks for coming back, I completely forgot about my comment.

`process.VtxSmeared` is part of `process.pgen`. So you will get wrong  parameters for vertex smearing. Using `DBrealisticHLLHC` should give you the proper parameter.

srimanob replied:
`process.pgen = cms.Sequence(cms.SequencePlaceholder("randomEngineStateProducer")+process.VertexSmearing+process.GenSmeared+process.GeneInfo+process.genJetMET, process.PPSTransportTask)`
and
`process.VertexSmearing = cms.Sequence(cms.SequencePlaceholder("VtxSmeared"))`

srimanob replied:
Also typo, it should be `BeamSpot`, not `Beamspot`. With typo, it will pick default BS of `'Gen'`

srimanob replied:
Fix in https://github.com/cms-sw/cmssw/pull/45685

AdrianoDee replied:
Great thanks!
